lib/ruby_indexer/lib/ruby_indexer/configuration.rb

```diff
@@ -158,6 +178,37 @@ def apply_config(config)
 
     private
 
+    sig { params(base_directory: String, excluded_pattern: String).returns(T::Array[String]) }
+    def indexable_included_file_patterns(base_directory = Dir.pwd, excluded_pattern = exclude_pattern)
+      flags = File::FNM_PATHNAME | File::FNM_EXTGLOB
+
+      # Escape File.fnmatch? wildcards in the directory
+      base_directory = base_directory.gsub(/[\\\{\}\[\]\*\?]/) do |reserved_glob_character|
+        "\\#{reserved_glob_character}"
+      end
+
+      dirs = Dir.glob(File.join(base_directory, "*/"), flags)
+        .reject do |dir|
+        next true if dir.end_with?("/./", "/../")
+        next true if File.fnmatch?(excluded_pattern, dir, flags)
+
+        symlink_excluded_or_infinite_loop?(base_directory, dir, excluded_pattern, flags)
+      end
+
+      dirs
+        .flat_map { |dir| indexable_included_file_patterns(dir, excluded_pattern) }
+        .unshift(base_directory)
+    end
+
+    sig { params(base_dir: String, current_dir: String, exclude_pattern: String, flags: Integer).returns(T::Boolean) }
+    def symlink_excluded_or_infinite_loop?(base_dir, current_dir, exclude_pattern, flags)
+      dir_realpath = File.realpath(current_dir)
+      File.symlink?(current_dir.chomp("/")) && (
```

Comment:
The chomp may not work for Windows paths. Or does it?

Reply:
Rubocop has had this code for 3 years, so I assume it works on Windows.

Reply:
So I asked a windows user what `Dir.pwd`, `File.join(Dir.pwd, 'foo', 'bar')`, and `File::ALT_SEPARATOR` return.

I expected the pwd to use back slashes, but it uses forward slashes. The file join also uses forward slashes. The alt separator is a backslash. So, the code we have here should be fine, since that's the same thing I see on my Mac.

Does Shopify have any developers running Windows that you could ask to run those same commands? I can read some Ruby C code, if we need. I do see that `File.join`'s documentation specifically says it joins using `"/"`. https://rubyapi.org/3.3/o/file#method-c-join

Reply:
Nope. It can't have a trailing slash.

```sh
$ ln -s test/fixtures/ test_symlink
```

```ruby
File.symlink?('test_symlink')
=> true

File.symlink?('test_symlink/')
=> false

File.symlink?('test_symlink/'.chomp('/'))
=> true
```